`test_git_sparse_paths_partial_clone` fails locally

### Steps to reproduce

```console
$ git --version
git version 2.25.1
$ spack unit-test lib/spack/spack/test/git_fetch.py::test_git_sparse_paths_partial_clone[None]
```

### Error message

```console
========================================================= test session starts ==========================================================
platform linux -- Python 3.11.5, pytest-8.3.2, pluggy-1.5.0
rootdir: /home/culpo/PycharmProjects/spack
configfile: pytest.ini
plugins: xdist-3.6.1
collected 1 item                                                                                                                       

lib/spack/spack/test/git_fetch.py F                                                                                              [100%]

=============================================================== FAILURES ===============================================================
______________________________________________ test_git_sparse_paths_partial_clone[None] _______________________________________________

mock_git_repository = <spack.util.pattern.Bunch object at 0x7ff95b5a1d10>, git_version = Version("2.25.1")
default_mock_concretization = <function default_mock_concretization.<locals>._func at 0x7ff95b5c63e0>
mutable_mock_repo = <spack.repo.RepoPath object at 0x7ff95bf3a6d0>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7ff95b5a30d0>

    @pytest.mark.disable_clean_stage_check
    def test_git_sparse_paths_partial_clone(
        mock_git_repository, git_version, default_mock_concretization, mutable_mock_repo, monkeypatch
    ):
        """
        Test partial clone of repository when using git_sparse_paths property
        """
        type_of_test = "many-directories"
        sparse_paths = ["dir0"]
        omitted_paths = ["dir1", "dir2"]
        t = mock_git_repository.checks[type_of_test]
        args = copy.copy(t.args)
        args["git_sparse_paths"] = sparse_paths
        s = default_mock_concretization("git-test")
        monkeypatch.setitem(s.package.versions, Version("git"), args)
        s.package.do_stage()
        with working_dir(s.package.stage.source_path):
            # top level directory files are cloned via sparse-checkout
>           assert os.path.isfile("r0_file")
E           AssertionError: assert False
E            +  where False = <function isfile at 0x7ff95ea8bce0>('r0_file')
E            +    where <function isfile at 0x7ff95ea8bce0> = <module 'posixpath' (frozen)>.isfile
E            +      where <module 'posixpath' (frozen)> = os.path

lib/spack/spack/test/git_fetch.py:413: AssertionError
-------------------------------------------------------- Captured stdout setup ---------------------------------------------------------
Initialized empty Git repository in /tmp/pytest-of-culpo/pytest-6/mock-git-repo-submodule-dir-00/spack-src/.git/
[master (root-commit) 0b2185a] mock-git-repo r0 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 r0_file_0
Initialized empty Git repository in /tmp/pytest-of-culpo/pytest-6/mock-git-repo-submodule-dir-10/spack-src/.git/
[master (root-commit) 4909a1c] mock-git-repo r0 1
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 r0_file_1
Initialized empty Git repository in /tmp/pytest-of-culpo/pytest-6/mock-git-repo-dir0/spack-src/.git/
[master (root-commit) 1af5b55] mock-git-repo r0
 4 files changed, 8 insertions(+)
 create mode 100644 .gitmodules
 create mode 100644 r0_file
 create mode 160000 third_party/submodule0
 create mode 160000 third_party/submodule1
[test-branch 2be024b] r1 test branch
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 branch_file
[tag-branch 9251da6] tag test branch
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 tag_file
[master 8a37158] mock-git-repo r2
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 r2_file
[many_dirs 531e97d] many_dirs add files
 6 files changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 dir0/file0
 create mode 100644 dir0/file1
 create mode 100644 dir1/file0
 create mode 100644 dir1/file1
 create mode 100644 dir2/file0
 create mode 100644 dir2/file1
-------------------------------------------------------- Captured stderr setup ---------------------------------------------------------
Cloning into '/tmp/pytest-of-culpo/pytest-6/mock-git-repo-dir0/spack-src/third_party/submodule0'...
Cloning into '/tmp/pytest-of-culpo/pytest-6/mock-git-repo-dir0/spack-src/third_party/submodule1'...
Switched to branch 'test-branch'
Switched to branch 'tag-branch'
Switched to branch 'master'
Switched to a new branch 'many_dirs'
Switched to branch 'master'
--------------------------------------------------------- Captured stderr call ---------------------------------------------------------
warning: filtering not recognized by server, ignoring
warning: filtering not recognized by server, ignoring
========================================================= slowest 30 durations =========================================================
0.41s call     lib/spack/spack/test/git_fetch.py::test_git_sparse_paths_partial_clone[None]
0.41s setup    lib/spack/spack/test/git_fetch.py::test_git_sparse_paths_partial_clone[None]

(1 durations < 0.005s hidden.  Use -vv to show these durations.)
======================================================= short test summary info ========================================================
FAILED lib/spack/spack/test/git_fetch.py::test_git_sparse_paths_partial_clone[None] - AssertionError: assert False
========================================================== 1 failed in 0.94s ===========================================================
```

### Information on your system

 **Spack:** 0.23.0.dev0 (1b82779087ed4bb7b90d30da02ffff9b655d6fcc)
* **Python:** 3.11.5
* **Platform:** linux-ubuntu20.04-icelake

@psakievich @becker33 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

--- a/lib/spack/spack/fetch_strategy.py
+++ b/lib/spack/spack/fetch_strategy.py
@@ -930,7 +930,9 @@ class GitFetchStrategy(VCSFetchStrategy):
         dest = self.stage.source_path
         git = self.git
 
-        if self.git_version < spack.version.Version("2.25.0.0"):
+        if self.git_version < spack.version.Version("2.26.0"):
+            # technically this should be supported for 2.25, but bumping for OS issues 
+            # see https://github.com/spack/spack/issues/45771
             # code paths exist where the package is not set.  Assure some indentifier for the
             # package that was configured  for sparse checkout exists in the error message
             identifier = str(self.url)

--- a/lib/spack/spack/fetch_strategy.py
+++ b/lib/spack/spack/fetch_strategy.py
@@ -931,7 +931,7 @@ class GitFetchStrategy(VCSFetchStrategy):
         git = self.git
 
         if self.git_version < spack.version.Version("2.26.0"):
-            # technically this should be supported for 2.25, but bumping for OS issues 
+            # technically this should be supported for 2.25, but bumping for OS issues
             # see https://github.com/spack/spack/issues/45771
             # code paths exist where the package is not set.  Assure some indentifier for the
             # package that was configured  for sparse checkout exists in the error message

--- a/lib/spack/spack/test/git_fetch.py
+++ b/lib/spack/spack/test/git_fetch.py
@@ -415,7 +415,7 @@ def test_git_sparse_paths_partial_clone(
         for p in sparse_paths:
             assert os.path.isdir(p)
 
-        if git_version < Version("2.25.0.0"):
+        if git_version < Version("2.26.0.0"):
             # older versions of git should fall back to a full clone
             for p in omitted_paths:
                 assert os.path.isdir(p)

